src/nvim/window.c

```diff
@@ -2973,6 +2973,7 @@ int win_new_tabpage(int after)
     return FAIL;
   }
   curtab = newtp;
+  apply_autocmds(EVENT_TABNEW, arg, arg, false, curbuf);
 
   /* Create a new empty window. */
   if (win_alloc_firstwin(tp->tp_curwin) == OK) {
```

Comment:
Uhm, okay. Switching `TabEnter` and `WinEnter` is worth an own PR.. `:h TabEnter` explicitly mentions that `TabEnter` comes after `WinEnter` and it breaks a test, too. I leave it away for now.